### DISCUSSION TITLE
Help setting up DNS records with pfSense

### DISCUSSION THREAD

**Discussion by jasonajack** [postalserver/postal#3253] [Help with using Postal] [answered]
Hello, I'm trying to configure Postal using local DNS via pfSense. My use case is to send emails from my local services through Postal to my gmail (for notifications, etc.)

I am not a DNS export, and have only gotten familiar with it through this exercise in trying to get my domain to work, but I cannot get past SPF validation.

I have tried the following in my pfSense "Custom options":

```
local-zone: 'postal.<homenetwork>.' static
local-data: 'postal.<homenetwork>. IN A 10.0.0.X'
local-data: 'mx.postal.<homenetwork>. IN A 10.0.0.X'
local-data: 'postal.<homenetwork>. IN MX 10 mx.postal.<homenetwork>.'
local-data: 'postal.<homenetwork>. IN TXT "v=spf1 a mx include:spf.postal.<homenetwork> ~all"'
```

I have tried several other configurations and cannot seem to get past the SPF validation check, and I do not know where I am going wrong. I have searched through several other issues and have not found any answer. Can someone please help?

**Comment by willpower232** [comment, marked as answer]
I don't know how that works in pfSense but presumably you could use `nslookup` on your Postal server to confirm it can receive those records.

If you aren't using a real domain name, and real publicly available DNS records, then I can imagine that your email may be blocked from gmail as it will look like a very fake email.

Also I believe that you don't need to get the DNS checks page to pass in order to send emails with Postal, the checks are mostly to ensure that other people can receive your email so cheating them with pfSense doesn't really help anything in my opinion.

**Reply by jasonajack** [reply]
> I don't know how that works in pfSense but presumably you could use `nslookup` on your Postal server to confirm it can receive those records.

Yeah, it all works locally but the check fails.

> If you aren't using a real domain name, and real publicly available DNS records, then I can imagine that your email may be blocked from gmail as it will look like a very fake email.
> 
> Also I believe that you don't need to get the DNS checks page to pass in order to send emails with Postal, the checks are mostly to ensure that other people can receive your email so cheating them with pfSense doesn't really help anything in my opinion.

You're probably right. I started from scratch and set up a new public subdomain using DuckDNS. They have a way to add TXT records via an API command. After reinitializing and configuring through their API I was able to get past the SPF check, so it seems that was the issue all along.

Thank you for your help!
